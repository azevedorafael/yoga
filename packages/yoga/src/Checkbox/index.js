import Switch from './web/Switch';

const Checkbox = {};

Checkbox.Switch = Switch;

export default Checkbox;
