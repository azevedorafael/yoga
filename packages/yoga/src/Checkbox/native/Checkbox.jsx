import React, { useState } from 'react';
import { bool, func, string } from 'prop-types';
import styled from 'styled-components';
import { TouchableWithoutFeedback, View } from 'react-native';
import CheckMark from './CheckMark';

const CheckboxWrapper = styled.View`
  flex-direction: row;
  align-items: center;
`;

const Label = styled.Text(
  ({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
  padding-left: ${checkbox.label.padding.left}px;

  font-size: ${checkbox.label.font.size}px;

  color: ${checkbox.label.font.color};
`,
);

const HelperWrapper = styled.View(
  ({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
  margin-top: ${checkbox.helper.margin.top}px;
`,
);

const Helper = styled.Text(
  ({
    error,
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
  font-size: ${checkbox.helper.font.size}px;

  color: ${
    error ? checkbox.helper.selected.font.color : checkbox.helper.font.color
  };
`,
);

const Checkbox = ({
  label,
  helper,
  disabled,
  checked,
  error,
  onPressIn,
  onPressOut,
  ...rest
}) => {
  const [pressed, setPressed] = useState(false);
  return (
    <View>
      <TouchableWithoutFeedback
        {...rest}
        disabled={disabled}
        onPressIn={e => {
          setPressed(true);
          onPressIn(e);
        }}
        onPressOut={e => {
          setPressed(false);
          onPressOut(e);
        }}
      >
        <CheckboxWrapper>
          <CheckMark
            disabled={disabled}
            checked={checked}
            error={error}
            pressed={pressed}
          />
          <Label>{label}</Label>
        </CheckboxWrapper>
      </TouchableWithoutFeedback>
      {helper && (
        <HelperWrapper>
          <Helper error={error}>{helper}</Helper>
        </HelperWrapper>
      )}
    </View>
  );
};

Checkbox.propTypes = {
  label: string,
  /** set a short helper text under checkbox */
  helper: string,
  checked: bool,
  disabled: bool,
  error: bool,
  onPressIn: func,
  onPressOut: func,
};

Checkbox.defaultProps = {
  label: undefined,
  helper: undefined,
  checked: false,
  disabled: false,
  error: false,
  onPressIn: () => {},
  onPressOut: () => {},
};

Checkbox.displayName = 'Checkbox';

export default Checkbox;
