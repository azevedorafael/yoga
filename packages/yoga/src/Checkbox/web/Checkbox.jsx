import React from 'react';
import { bool, string, objectOf, any } from 'prop-types';
import styled from 'styled-components';
import { hexToRgb } from '@gympass/yoga-common';
import { Done } from '@gympass/yoga-icons';

import { HiddenInput } from '../../shared';

const CheckboxWrapper = styled.div`
  display: inline-block;

  * {
    box-sizing: border-box;
  }
`;

const CheckMark = styled.div`
  position: relative;

  border-style: solid;

  ${({
    checked,
    disabled,
    error,
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
    width: ${checkbox.size}px;
    height: ${checkbox.size}px;

    margin-right: ${checkbox.margin.right}px;

    border-radius: ${checkbox.border.radius}px;
    border-width: ${checkbox.border.width}px;

    border-color: ${
      disabled ? checkbox.disabled.border.color : checkbox.border.color
    };

    ${
      checked
        ? `
        background-color: ${checkbox.checked.backgroundColor}; 

        svg {
          position: absolute;
          top: 50%;
          left: 50%;
          
          fill: ${checkbox.checked.icon.color};

          transform: translate(-50%, -50%);
        }
        `
        : ''
    }

    ${
      disabled && checked
        ? `background-color: ${checkbox.disabled.backgroundColor};`
        : ''
    }

    ${error ? `border-color: ${checkbox.error.border.color};` : ''}

    ${
      error && checked
        ? `background-color: ${checkbox.error.backgroundColor};`
        : ''
    }
  `}
`;

const Label = styled.label`
  display: flex;
  flex-direction: row;
  align-items: center;

  cursor: pointer;

  ${({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
    font-size: ${checkbox.label.font.size}px;
    color: ${checkbox.label.font.color};
  `}
`;

const Shadow = styled.span`
  position: absolute;
  display: none;

  ${({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
    width: ${checkbox.size}px;
    height: ${checkbox.size}px;
    border-radius: ${checkbox.hover.border.radius}px;
  `}
`;

const CheckboxStyled = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  ${({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `

  ${Label}:active,
  &:focus-within,
  &:hover  {
    ${Shadow} {
      display: block;
    }
  }

  ${Label}:active {
    ${Shadow} {
      background-color: ${hexToRgb(checkbox.hover.backgroundColor, 0.75)};

      box-shadow: 0 0 0 ${Math.round(
        checkbox.size * 0.33 + checkbox.border.width * 2,
      )}px
      ${hexToRgb(checkbox.hover.backgroundColor, 0.75)};
    }
  }

  &:focus-within {
    ${Shadow} {
      background-color: ${hexToRgb(checkbox.hover.backgroundColor, 0.5)};
      
      box-shadow: 0 0 0 ${Math.round(
        checkbox.size * 0.33 + checkbox.border.width * 2,
      )}px
        ${hexToRgb(checkbox.hover.backgroundColor, 0.5)};
      }
  }

  &:hover {
    ${Shadow} {
      background-color: ${hexToRgb(checkbox.hover.backgroundColor, 0.25)};

      box-shadow: 0 0 0 ${Math.round(
        checkbox.size * 0.33 + checkbox.border.width * 2,
      )}px
      ${hexToRgb(checkbox.hover.backgroundColor, 0.25)};
    }
  }
  `}
`;

const HelperWrapper = styled.div`
  width: 100%;

  ${({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
    margin-top: ${checkbox.helper.margin.top}px;
  `}
`;

const Helper = styled.span`
  ${({
    error,
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
    color: ${
      error ? checkbox.helper.selected.font.color : checkbox.helper.font.color
    };

    font-size: ${checkbox.helper.font.size}px;
  `}
`;

const Checkbox = ({
  value,
  label,
  helper,
  disabled,
  checked,
  error,
  style,
  className,
  ...rest
}) => (
  <CheckboxWrapper style={style} className={className}>
    <CheckboxStyled>
      <Label>
        <Shadow />
        <CheckMark disabled={disabled} checked={checked} error={error}>
          {checked && <Done />}
        </CheckMark>
        <HiddenInput
          type="checkbox"
          checked={checked}
          disabled={disabled}
          {...rest}
        />
        {label}
      </Label>
    </CheckboxStyled>
    {helper && (
      <HelperWrapper>
        <Helper error={error}>{helper}</Helper>
      </HelperWrapper>
    )}
  </CheckboxWrapper>
);

Checkbox.propTypes = {
  label: string,
  /** short helper text under checkbox */
  helper: string,
  value: string,
  checked: bool,
  disabled: bool,
  error: bool,
  /** set a style to the checkbox container */
  style: objectOf(any),
  className: string,
};

Checkbox.defaultProps = {
  label: undefined,
  value: undefined,
  helper: undefined,
  checked: false,
  disabled: false,
  error: false,
  style: undefined,
  className: undefined,
};

Checkbox.displayName = 'Checkbox';

export default Checkbox;
