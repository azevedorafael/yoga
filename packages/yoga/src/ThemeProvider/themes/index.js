import corp from './corp';
import endUser from './end-user';
import gyms from './gyms';

export { endUser as default, endUser, corp, gyms };
