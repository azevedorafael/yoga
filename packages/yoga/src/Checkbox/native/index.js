import Switch from './Switch';
import Checkbox from './Checkbox';

export { Checkbox, Switch };
