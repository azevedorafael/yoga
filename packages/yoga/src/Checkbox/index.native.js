import { Checkbox, Switch } from './native';

Checkbox.Switch = Switch;

export default Checkbox;
