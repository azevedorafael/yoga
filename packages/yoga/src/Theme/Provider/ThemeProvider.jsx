import React from 'react';
import { ThemeProvider as SCThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';
import * as tokens from '@gympass/yoga-tokens';
import * as themes from '../themes';

const getTheme = ({ theme, locale }) => {
  const token = tokens[locale] || tokens.default;
  const appTheme = themes[theme] || themes.default;

  return appTheme(token);
};

const ThemeProvider = ({ children, ...theme }) => (
  <SCThemeProvider theme={{ yoga: getTheme(theme) }}>
    {children}
  </SCThemeProvider>
);

ThemeProvider.propTypes = {
  theme: PropTypes.string,
  locale: PropTypes.string,
  children: PropTypes.node.isRequired,
};

ThemeProvider.defaultProps = {
  theme: 'endUser',
  locale: 'pt-BR',
};

export default ThemeProvider;
