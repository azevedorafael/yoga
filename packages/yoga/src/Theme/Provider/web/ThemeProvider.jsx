import React from 'react';
import { node } from 'prop-types';

import Provider from '../Provider';
import GlobalStyle from './GlobalStyle';

/** This component provides a theme to all React components underneath itself via the context API. */
const ThemeProvider = ({ children, ...rest }) => (
  <Provider {...rest}>
    <GlobalStyle />
    {children}
  </Provider>
);

ThemeProvider.propTypes = {
  children: node.isRequired,
};

export default ThemeProvider;
