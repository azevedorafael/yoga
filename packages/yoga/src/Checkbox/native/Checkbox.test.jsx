import React from 'react';
import { render, fireEvent, toJSON } from '@testing-library/react-native';

import Checkbox from '..';
import ThemeProvider from '../../Theme';

const data = {
  value: 'value',
  label: 'Checkbox Component',
  helper: 'Helper Text',
};

describe('<Checkbox />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with default checkbox', () => {
      const { container } = render(
        <ThemeProvider>
          <Checkbox {...data} />
        </ThemeProvider>,
      );
      expect(toJSON(container)).toMatchSnapshot();
    });

    it('should match snapshot with disabled checkbox', () => {
      const { container } = render(
        <ThemeProvider>
          <Checkbox {...data} disabled />
        </ThemeProvider>,
      );
      expect(toJSON(container)).toMatchSnapshot();
    });

    it('should match snapshot with checked checkbox', () => {
      const { container } = render(
        <ThemeProvider>
          <Checkbox {...data} checked />
        </ThemeProvider>,
      );
      expect(toJSON(container)).toMatchSnapshot();
    });

    it('should match snapshot with disabled and checked checkbox', () => {
      const { container } = render(
        <ThemeProvider>
          <Checkbox {...data} disabled checked />
        </ThemeProvider>,
      );
      expect(toJSON(container)).toMatchSnapshot();
    });

    it('should match snapshot with error checkbox', () => {
      const { container } = render(
        <ThemeProvider>
          <Checkbox {...data} error />
        </ThemeProvider>,
      );
      expect(toJSON(container)).toMatchSnapshot();
    });

    it('should match snapshot with error and checked checkbox', () => {
      const { container } = render(
        <ThemeProvider>
          <Checkbox {...data} error checked />
        </ThemeProvider>,
      );
      expect(toJSON(container)).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('should call onChange function when press on Checkbox', () => {
      const onChangeMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Checkbox {...data} onPress={onChangeMock} testID="checkbox" />
        </ThemeProvider>,
      );
      fireEvent.press(getByTestId('checkbox'));
      expect(onChangeMock).toHaveBeenCalled();
    });
  });
});
