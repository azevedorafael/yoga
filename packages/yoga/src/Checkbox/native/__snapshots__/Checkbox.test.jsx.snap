// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Snapshots should match snapshot with checked checkbox 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    accessibilityRole="checkbox"
    disabled={false}
    value="value"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "width": "100%",
          },
        ]
      }
    >
      <View
        checked={true}
        disabled={false}
        error={false}
        style={
          Array [
            Object {
              "backgroundColor": "#F46152",
              "borderColor": "#F46152",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "width": 24,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAQAAAAAYLlVAAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAAAmJLR0QAAKqNIzIAAAAJcEhZcwAADsQAAA7EAZUrDhsAAAAHdElNRQfjCw4QFioQDlrfAAABI0lEQVRo3u2YPW7CQBBGP18A0UXpaZEQJ+AG6VJyEroIChrEUSKROkWOkN+WHiSgRnp0QDCQtdeejZL5+tn31tLOjCx5PJ7/niwNlrZ6Wukx26TBD9gCMKeVAj/ikKe0eFinxcNHWjz0LfEPOfzY8Y53vOMd73jHO/534LnnhVcmNCLw+WVrGFra35e8c2OOl/g8KiulEIWXWH8rLawQiZeYnZQXUojGS7RYllWoAC9JdFicHPPFbUBdde++jELFbaeoQg1dr4hCTU03VKHGnh+iUPPI+UnBYOJdUzAauJcUDOc93TPdcZrDl+l6EV/B6vaBChbL1hUFq13vgoLpqplXsMSfUbDGSxJd5gBsGVR/etDvehq6U1PP2VuC+3s8nj+fHXfWlhGVw0EjAAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDE5LTExLTE0VDE2OjIyOjQyKzAwOjAwR6ZpUQAAACV0RVh0ZGF0ZTptb2RpZnkAMjAxOS0xMS0xNFQxNjoyMjo0MiswMDowMDb70e0AAAAZdEVYdFNvZnR3YXJlAHd3dy5pbmtzY2FwZS5vcmeb7jwaAAAAAElFTkSuQmCC",
            }
          }
          style={
            Array [
              Object {
                "height": 11,
                "marginBottom": 4,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 4,
                "width": 16,
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontSize": 14,
              "paddingLeft": 8,
            },
          ]
        }
      >
        Checkbox Component
      </Text>
    </View>
  </TouchableWithoutFeedback>
  <View
    style={
      Array [
        Object {
          "flexWrap": "nowrap",
          "marginTop": 4,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      error={false}
      style={
        Array [
          Object {
            "color": "#9898A6",
            "fontSize": 14,
          },
        ]
      }
    >
      Helper Text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot with default checkbox 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    accessibilityRole="checkbox"
    disabled={false}
    value="value"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "width": "100%",
          },
        ]
      }
    >
      <View
        checked={false}
        disabled={false}
        error={false}
        style={
          Array [
            Object {
              "borderColor": "#F46152",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "width": 24,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontSize": 14,
              "paddingLeft": 8,
            },
          ]
        }
      >
        Checkbox Component
      </Text>
    </View>
  </TouchableWithoutFeedback>
  <View
    style={
      Array [
        Object {
          "flexWrap": "nowrap",
          "marginTop": 4,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      error={false}
      style={
        Array [
          Object {
            "color": "#9898A6",
            "fontSize": 14,
          },
        ]
      }
    >
      Helper Text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot with disabled and checked checkbox 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    accessibilityRole="checkbox"
    disabled={true}
    value="value"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "width": "100%",
          },
        ]
      }
    >
      <View
        checked={true}
        disabled={true}
        error={false}
        style={
          Array [
            Object {
              "backgroundColor": "#9898A6",
              "borderColor": "#9898A6",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "width": 24,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAQAAAAAYLlVAAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAAAmJLR0QAAKqNIzIAAAAJcEhZcwAADsQAAA7EAZUrDhsAAAAHdElNRQfjCw4QFioQDlrfAAABI0lEQVRo3u2YPW7CQBBGP18A0UXpaZEQJ+AG6VJyEroIChrEUSKROkWOkN+WHiSgRnp0QDCQtdeejZL5+tn31tLOjCx5PJ7/niwNlrZ6Wukx26TBD9gCMKeVAj/ikKe0eFinxcNHWjz0LfEPOfzY8Y53vOMd73jHO/534LnnhVcmNCLw+WVrGFra35e8c2OOl/g8KiulEIWXWH8rLawQiZeYnZQXUojGS7RYllWoAC9JdFicHPPFbUBdde++jELFbaeoQg1dr4hCTU03VKHGnh+iUPPI+UnBYOJdUzAauJcUDOc93TPdcZrDl+l6EV/B6vaBChbL1hUFq13vgoLpqplXsMSfUbDGSxJd5gBsGVR/etDvehq6U1PP2VuC+3s8nj+fHXfWlhGVw0EjAAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDE5LTExLTE0VDE2OjIyOjQyKzAwOjAwR6ZpUQAAACV0RVh0ZGF0ZTptb2RpZnkAMjAxOS0xMS0xNFQxNjoyMjo0MiswMDowMDb70e0AAAAZdEVYdFNvZnR3YXJlAHd3dy5pbmtzY2FwZS5vcmeb7jwaAAAAAElFTkSuQmCC",
            }
          }
          style={
            Array [
              Object {
                "height": 11,
                "marginBottom": 4,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 4,
                "width": 16,
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontSize": 14,
              "paddingLeft": 8,
            },
          ]
        }
      >
        Checkbox Component
      </Text>
    </View>
  </TouchableWithoutFeedback>
  <View
    style={
      Array [
        Object {
          "flexWrap": "nowrap",
          "marginTop": 4,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      error={false}
      style={
        Array [
          Object {
            "color": "#9898A6",
            "fontSize": 14,
          },
        ]
      }
    >
      Helper Text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot with disabled checkbox 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    accessibilityRole="checkbox"
    disabled={true}
    value="value"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "width": "100%",
          },
        ]
      }
    >
      <View
        checked={false}
        disabled={true}
        error={false}
        style={
          Array [
            Object {
              "borderColor": "#9898A6",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "width": 24,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontSize": 14,
              "paddingLeft": 8,
            },
          ]
        }
      >
        Checkbox Component
      </Text>
    </View>
  </TouchableWithoutFeedback>
  <View
    style={
      Array [
        Object {
          "flexWrap": "nowrap",
          "marginTop": 4,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      error={false}
      style={
        Array [
          Object {
            "color": "#9898A6",
            "fontSize": 14,
          },
        ]
      }
    >
      Helper Text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot with error and checked checkbox 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    accessibilityRole="checkbox"
    disabled={false}
    value="value"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "width": "100%",
          },
        ]
      }
    >
      <View
        checked={true}
        disabled={false}
        error={true}
        style={
          Array [
            Object {
              "backgroundColor": "#CB3530",
              "borderColor": "#CB3530",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "width": 24,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAQAAAAAYLlVAAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAAAmJLR0QAAKqNIzIAAAAJcEhZcwAADsQAAA7EAZUrDhsAAAAHdElNRQfjCw4QFioQDlrfAAABI0lEQVRo3u2YPW7CQBBGP18A0UXpaZEQJ+AG6VJyEroIChrEUSKROkWOkN+WHiSgRnp0QDCQtdeejZL5+tn31tLOjCx5PJ7/niwNlrZ6Wukx26TBD9gCMKeVAj/ikKe0eFinxcNHWjz0LfEPOfzY8Y53vOMd73jHO/534LnnhVcmNCLw+WVrGFra35e8c2OOl/g8KiulEIWXWH8rLawQiZeYnZQXUojGS7RYllWoAC9JdFicHPPFbUBdde++jELFbaeoQg1dr4hCTU03VKHGnh+iUPPI+UnBYOJdUzAauJcUDOc93TPdcZrDl+l6EV/B6vaBChbL1hUFq13vgoLpqplXsMSfUbDGSxJd5gBsGVR/etDvehq6U1PP2VuC+3s8nj+fHXfWlhGVw0EjAAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDE5LTExLTE0VDE2OjIyOjQyKzAwOjAwR6ZpUQAAACV0RVh0ZGF0ZTptb2RpZnkAMjAxOS0xMS0xNFQxNjoyMjo0MiswMDowMDb70e0AAAAZdEVYdFNvZnR3YXJlAHd3dy5pbmtzY2FwZS5vcmeb7jwaAAAAAElFTkSuQmCC",
            }
          }
          style={
            Array [
              Object {
                "height": 11,
                "marginBottom": 4,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 4,
                "width": 16,
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontSize": 14,
              "paddingLeft": 8,
            },
          ]
        }
      >
        Checkbox Component
      </Text>
    </View>
  </TouchableWithoutFeedback>
  <View
    style={
      Array [
        Object {
          "flexWrap": "nowrap",
          "marginTop": 4,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      error={true}
      style={
        Array [
          Object {
            "color": "#CB3530",
            "fontSize": 14,
          },
        ]
      }
    >
      Helper Text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot with error checkbox 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    accessibilityRole="checkbox"
    disabled={false}
    value="value"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "width": "100%",
          },
        ]
      }
    >
      <View
        checked={false}
        disabled={false}
        error={true}
        style={
          Array [
            Object {
              "borderColor": "#CB3530",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "width": 24,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontSize": 14,
              "paddingLeft": 8,
            },
          ]
        }
      >
        Checkbox Component
      </Text>
    </View>
  </TouchableWithoutFeedback>
  <View
    style={
      Array [
        Object {
          "flexWrap": "nowrap",
          "marginTop": 4,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      error={true}
      style={
        Array [
          Object {
            "color": "#CB3530",
            "fontSize": 14,
          },
        ]
      }
    >
      Helper Text
    </Text>
  </View>
</View>
`;
